hackathon_site/dashboard/frontend/src/components/teamDetail/ProjectDescription/ProjectDescription.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import { Formik, Form, Field, FormikValues } from "formik";
+import * as Yup from "yup";
+import { TextField, Button, Box, Grid } from "@material-ui/core";
+import { useDispatch } from "react-redux";
+import { updateProjectDescription } from "slices/event/teamDetailSlice";
+
+interface ProjectDescriptionProps {
+    teamCode: string;
+}
+
+const ProjectDescription = ({ teamCode }: ProjectDescriptionProps) => {
+    const dispatch = useDispatch();
+    const initialProjectDescription = "Write your project description here";
+    const [isEditing, setIsEditing] = useState(false);
+    const projectDescriptionSchema = Yup.object().shape({
+        projectDescription: Yup.string()
+            .max(500)
+            .required("Project description is required"),
+    });
+
+    const handleSubmit = async (values: FormikValues, { setSubmitting }: any) => {
+        try {
+            await dispatch(
+                updateProjectDescription({
+                    teamCode: teamCode,
+                    projectDescription: values.projectDescription,
+                })
+            );
+        } catch (error) {
+        } finally {
+            setSubmitting(false);
+        }
+    };
+
+    return (
+        <div>
+            <Formik
+                initialValues={{ projectDescription: initialProjectDescription }}
+                validationSchema={projectDescriptionSchema}
+                onSubmit={handleSubmit}
+            >
+                {({ isSubmitting, isValid }) => (
+                    <Form>
+                        <Field
+                            as={TextField}
+                            name="projectDescription"
+                            multiline
+                            fullWidth
+                            variant="outlined"
+                            disabled={!isEditing}
+                            rows={4}
+                            style={{ margin: "20px 5px 0 0" }}
```

Comment:
can you put the styling in this file in `ProjectDescription.module.scss`. Reminder for scss you need to compile in order for the css to show, refer to README.md on how to do that.